feat(codepipeline-actions): support `InspectorEcrImageScanAction` and `InspectorSourceCodeScanAction` actions

go-to-k opened the pull request:
### Issue # (if applicable)

Closes #33377.

### Reason for this change

<!--What is the bug or use case behind this change?-->

AWS CodePipeline introduces new build action: InspectorScan action.

- What's new: https://aws.amazon.com/jp/about-aws/whats-new/2024/11/aws-codepipeline-publishing-ecr-image-aws-inspectorscan-actions/
- Ref: https://docs.aws.amazon.com/codepipeline/latest/userguide/action-reference-InspectorScan.html
- IAM management: https://docs.aws.amazon.com/codepipeline/latest/userguide/security-iam.html#how-to-custom-role

### Description of changes

<!--
What code changes did you make? 
Have you made any important design decisions?
What AWS use cases does this change enable? To enable the use cases, which AWS service features are utilized?
-->

Add `InspectorEcrImageScanAction` and `InspectorSourceCodeScanAction` classes that extends `BaseInspectorScan` class.

The `BaseInspectorScan` class that extends `Action` class to aws-codepipeline-actions module.

### Describe any new or updated permissions being added

<!-- What new or updated IAM permissions are needed to support the changes being introduced ? -->


### Description of how you validated changes

<!--Have you added any unit tests and/or integration tests?-->

Both unit and integ tests

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation reviewed (changes requested):
The pull request linter fails with the following errors:

    ❌ Features must contain a change to a README file.
    ❌ Features must contain a change to a test file.
    ❌ Features must contain a change to an integration test file and the resulting snapshot.

If you believe this pull request should receive an exemption, please comment and provide a justification. A comment requesting an exemption should contain the text `Exemption Request`. Additionally, if clarification is needed, add `Clarification Request` to a comment.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: b65a7ec789dc7520f8fa3abdf0bf73f882855711
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=3b55975a-6fa4-4504-a390-3207aad81d51%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 52b96ee5398d68dc49a73f6c9421c243089cb8a9
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b498d920-f09b-4efb-af16-fa50e6f74945%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 3ec6f772d6f625e99f7273ce208e4c6fbfe4726d
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=04abfcc5-270e-4c5b-be40-f5659d1ba245%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 3624f586f2adc36a64b44b8aa031b1fe99fd279c
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=93b76792-6158-411f-a58a-0ccd3f35b918%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: c6ab25cf37507dce31e79943a983ff85719abf46
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=92cc029f-e610-41e7-82ba-8c55bebb1554%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: ea99afa1c02b118059e2c080d81e1a13258b6c08
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bf2cfd62-3f2f-4cc1-a8f3-90b186eab619%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

go-to-k commented on packages/aws-cdk-lib/aws-codepipeline-actions/lib/inspector/ecr-image-scan-action.ts:
```diff
@@ -0,0 +1,65 @@
+  codepipeline.ActionConfig {
+    const config = super.bound(scope, stage, options);
+
+    options.role.addToPrincipalPolicy(new iam.PolicyStatement({
+      resources: ['*'],
+      actions: [
+        'ecr:GetAuthorizationToken',
+      ],
+    }));
```
It wasn't in the documentation, but without it, the action gave an error.

```
An error occurred (AccessDeniedException) when calling the GetAuthorizationToken operation: User: arn:aws:sts::531991745243:assumed-role/codepipeline-ecr-image-sc-PipelineInvokeInspectorEc-z3v20FaFmI68/1739278219035 is not authorized to perform: ecr:GetAuthorizationToken on resource: * because no identity-based policy allows the ecr:GetAuthorizationToken action
Error: Cannot perform an interactive login from a non TTY device
```

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 05f41b5388913c525ec5ab70c1d9a5674a5c90e4
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=151bfd23-4057-442c-b2a6-d273a4f3dc2f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 41afadb305ed7f1af8dda24ebd03ca12449e2f97
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f8c037e8-f4a0-4d3f-96cd-cdaadb01de34%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: ad0f227a056d0fb6d5597144e903b7ffb3aa9882
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e8173438-8baa-4bb6-925b-5c90069f44c2%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: be4da86ce86824be553e6d8b009bd3ce1940ec31
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=63182eb2-dbb6-425d-98ec-8d3f042d54ba%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 90643d95a6e6b82a5f5dfe3b9255be5c4cf437a1
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6d4ee188-4f50-40b2-b693-8d9c685a2f03%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: e545b9c63ebb45812cd9ceb65f23269313450f0a
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a68a5fda-ed90-4c85-beec-c7eee81aa2fa%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

go-to-k commented on packages/aws-cdk-lib/aws-codepipeline-actions/test/inspector/inspector-ecr-image-scan-action.test.ts:
```diff
@@ -0,0 +1,320 @@
+    });
+  });
+
+  test('can get variables', () => {
+    // WHEN
+    const inspectorEcrImageScanAction = new cpactions.InspectorEcrImageScanAction({
+      actionName: 'InspectorScan',
+      output: scanOutput,
+      repository,
+    });
+
+    // THEN
+    expect(inspectorEcrImageScanAction.variables.highestScannedSeverity).toMatch(/^#{\${Token\[TOKEN\.[0-9]*\]}.HighestScannedSeverity}$/);
+  });
```
This and other common code are about the common behavior of the base class.

But as users are not aware of the base class because they call the subclasses directly, and in case there are any changes in the subclasses by contributors, it is also confirmation for regressions in the subclasses. So the tests are the same in `inspector-ecr-image-scan-action.test.ts` as in `inspector- ecr-image-scan-action.test.ts` to have the same content.

This also makes the test useful even if the base class is deleted and each class is implemented independently in the future.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 6a415d829213604b7ef8a86a341e3ea0d8bb97c8
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6758c239-ebd3-47e9-9a99-0f95333eacea%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation reviewed (changes requested):
The pull request linter fails with the following errors:

    ❌ CodeCov is indicating a drop in code coverage

If you believe this pull request should receive an exemption, please comment and provide a justification. A comment requesting an exemption should contain the text `Exemption Request`. Additionally, if clarification is needed, add `Clarification Request` to a comment.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: fab68f73d1cc44772c1ec1693e92b9150e8376a3
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2a973cc1-a571-41cf-a6bc-c2ac1efcc30e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 58a1d00a8b8f357b69063c79f3f5ed90b80eaef6
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e5d6246d-326a-4292-9670-a8ecb7dfdc7d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation reviewed (dismissed):
(This review is outdated)

go-to-k commented:
Exemption Request:

> ❌ CodeCov is indicating a drop in code coverage

Should already have written unit tests that cover the code I have added.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 202ffcd3bafcedf2177d0a7bb3ebc616dfa375ce
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a3737a77-bd02-4bb5-b7b7-8c003990a4cf%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation reviewed (dismissed):
(This review is outdated)

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 779d8e35575b45ed9e6743c0a01a3977b6563450
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c1a524d5-2e85-44c1-9d5d-d50abc963e7b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

go-to-k commented on packages/aws-cdk-lib/aws-codepipeline-actions/lib/inspector/scan-action-base.ts:
```diff
@@ -0,0 +1,130 @@
+  /**
+   * The highest severity output from the scan.
+   *
+   * Valid values are medium | high | critical.
+   */
+  readonly highestScannedSeverity: string;
```
This is not input to the construct, but output from the pipeline, which is a string formatted with the `variableExpression` method. Therefore, it cannot be an enum.

go-to-k commented on packages/aws-cdk-lib/aws-codepipeline-actions/lib/inspector/ecr-image-scan-action.ts:
```diff
@@ -0,0 +1,68 @@
+    // This permission was not listed in the above reference, but without it,
+    // an error would occur since `ecr get-login-password` is executed in the action.
+    options.role.addToPrincipalPolicy(new iam.PolicyStatement({
+      resources: ['*'],
```
The error message by CFn requested a wildcard.

go-to-k commented on packages/aws-cdk-lib/aws-codepipeline-actions/lib/inspector/ecr-image-scan-action.ts:
```diff
@@ -0,0 +1,68 @@
+      ],
+    }));
+
+    // This permission was not listed in the above reference, but without it,
+    // an error would occur since `ecr get-login-password` is executed in the action.
+    options.role.addToPrincipalPolicy(new iam.PolicyStatement({
+      resources: ['*'],
+      actions: [
+        'ecr:GetAuthorizationToken',
+      ],
```
The error message by CFn requested a wildcard.

5d reviewed (dismissed):
LGTM

5d commented:
Great work! 👍 Thanks

go-to-k commented:
@mergifyio update

go-to-k commented:
@5d 
Thanks for your approval!

But the merge process doesn't seem to be running. Could you please address this?

shikha372 commented:
@mergifyio update

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: a832486084aaeb8a1f1cce5da23e8f84b1921eb4
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d41d6e5e-9c4b-4966-b037-8bb362d2b51c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 81c8c5ceeb9e0b6c693a5c6091be7b357ac52e48
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b5d0faf4-ece4-48a5-95b0-8171c6cedb42%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: b6edaf87a0b53895b0c398ade5c476cd404340b4
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7ccd09c3-ab1c-434e-a6e3-c88a7a7130c5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: bdb589de2fa4852bb5d8faf2c7353305e81fef13
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b6178ee7-4765-40ab-8b42-10c10b2b7f59%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

go-to-k commented:
@5d @shikha372 

I updated the branch locally to address the mergify error.

In addition, the following changes were made:

- The link on the cp-action's reference page seems to have changed in the last few days, so I changed that as well.
  - commit: https://github.com/aws/aws-cdk/pull/33378/commits/6cfd08c6dde334320dd6cca1a31c2edb716d9820
- The already existing method for `ecr:GetAuthorizationToken` is now used.
  - commit: https://github.com/aws/aws-cdk/pull/33378/commits/81c8c5ceeb9e0b6c693a5c6091be7b357ac52e48
- A vulnerability was found in the Dockerfile for the integ test and the scan failed, so the threshold in the test was changed to avoid the error.
  - commit: https://github.com/aws/aws-cdk/pull/33378/commits/b6edaf87a0b53895b0c398ade5c476cd404340b4
- Since the feature flag (`@aws-cdk/pipelines:reduceStageRoleTrustScope`) was introduced in this [PR](https://github.com/aws/aws-cdk/pull/33742), changes to the snapshot will occur unless specified in the integ tests. So specified the flag in the integ tests.
  - commit: https://github.com/aws/aws-cdk/pull/33378/commits/bdb589de2fa4852bb5d8faf2c7353305e81fef13

Could you please check it again and merge it if you are okay?

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 231c07491f6480fd0230378a1f2c92d618d56173
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c14a8814-3dc0-463d-a849-ef7238ce5117%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->
